internal/metrics/telemetry/cluster_log_forwarder.go

```diff
@@ -1,76 +1,55 @@
 package telemetry
 
 import (
-	"strconv"
-
-	log "github.com/ViaQ/logerr/v2/log/static"
-	logging "github.com/openshift/cluster-logging-operator/api/logging/v1"
+	loggingv1 "github.com/openshift/cluster-logging-operator/api/logging/v1"
 	"github.com/openshift/cluster-logging-operator/internal/constants"
 )
 
-func UpdateInfofromCLF(forwarder logging.ClusterLogForwarder) {
+func (t *telemetryCollector) updateDefaultInfo(forwarder *loggingv1.ClusterLogForwarder) {
+	if forwarder.Namespace != constants.OpenshiftNS || forwarder.Name != constants.SingletonName {
```

Comment:
The code provides metrics for all the CLF instances.

This function is a separate codepath for the "default/legacy instance", because it might not even exist (in the case when a user has ClusterLogging without any customization in ClusterLogForwarder). So this function collects the final state of the "legacy instance" to expose it in metrics if the ClusterLogging instance exists while for all other CLF instances the information is directly gathered from the resource ([code here](https://github.com/openshift/cluster-logging-operator/pull/2443/files#diff-42a088a3e1209d9fa9065286d4c7b6b51a835fb67297f8d4b0976c2a2dea0bddR114)).

Reply:
Thinking about this again, I might have forgotten to describe this in more detail in a comment. I can add that to this function, if you like.

Reply:
May be reasonable to clarify though I'm the only one asking so maybe limited value.  This is not a blocking issue for me

Reply:
Added an explanation comment to the function. :heavy_check_mark: